pkg/proxy/ipvs/proxier.go

```diff
@@ -1944,59 +1945,58 @@ func (proxier *Proxier) deleteEndpointConnections(connectionMap []proxy.ServiceE
 	}
 }
 
-func (proxier *Proxier) syncService(svcName string, vs *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) error {
-	appliedVirtualServer, _ := proxier.ipvs.GetVirtualServer(vs)
-	if appliedVirtualServer == nil || !appliedVirtualServer.Equal(vs) {
-		if appliedVirtualServer == nil {
+func (proxier *Proxier) syncService(svcName string, updatedVS, currentVS *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) (*utilipvs.VirtualServer, error) {
+	if currentVS == nil || !currentVS.Equal(updatedVS) {
+		if currentVS == nil {
 			// IPVS service is not found, create a new service
-			klog.V(3).Infof("Adding new service %q %s:%d/%s", svcName, vs.Address, vs.Port, vs.Protocol)
-			if err := proxier.ipvs.AddVirtualServer(vs); err != nil {
+			klog.V(3).Infof("Adding new service %q %s:%d/%s", svcName, updatedVS.Address, updatedVS.Port, updatedVS.Protocol)
+			if err := proxier.ipvs.AddVirtualServer(updatedVS); err != nil {
 				klog.Errorf("Failed to add IPVS service %q: %v", svcName, err)
-				return err
+				return nil, err
 			}
 		} else {
 			// IPVS service was changed, update the existing one
 			// During updates, service VIP will not go down
 			klog.V(3).Infof("IPVS service %s was changed", svcName)
-			if err := proxier.ipvs.UpdateVirtualServer(vs); err != nil {
+			if err := proxier.ipvs.UpdateVirtualServer(updatedVS); err != nil {
 				klog.Errorf("Failed to update IPVS service, err:%v", err)
-				return err
+				return nil, err
 			}
 		}
+		var err error
+		currentVS, err = proxier.ipvs.GetVirtualServer(updatedVS)
```

Comment:
Can we define a new variable for currentVS instead of re-using the parameter name?

Reply:
currentVS is now a local variable since the currentIPVSServices map was moved to the Proxier struct.